### PR TITLE
Improve type join for function types.

### PR DESCRIPTION
**Pull request by rudkx**

### CODE REVIEW

**Comment by rudkx** [comment]
@swift-ci Please smoke test

File: lib/AST/TypeJoinMeet.cpp [hunk 1]
```diff
@@ -301,14 +304,23 @@ CanType TypeJoin::visitDependentMemberType(CanType second) {
 CanType TypeJoin::visitFunctionType(CanType second) {
   assert(First != second);
 
-  if (First->getKind() != second->getKind())
-    return TheAnyType;
+  auto secondFnTy = second->castTo<FunctionType>();
+
+  if (First->getKind() != second->getKind()) {
+    if (secondFnTy->getExtInfo().isNoEscape()) {
+      return Nonexistent;
```
Comment (slavapestov) [inline lib/AST/TypeJoinMeet.cpp]: Why not just return CanType()?

File: lib/Sema/MiscDiagnostics.cpp [hunk 2]
```diff
@@ -714,6 +715,8 @@ static void diagSyntacticUseRestrictions(TypeChecker &TC, const Expr *E,
           return;
         } else if (isa<MakeTemporarilyEscapableExpr>(parent)) {
           return;
+        } else if (isa<DynamicTypeExpr>(parent)) {
```
Comment (slavapestov) [inline lib/Sema/MiscDiagnostics.cpp]: Test case for this change?

Comment (rudkx) [reply lib/Sema/MiscDiagnostics.cpp]: Almost all of the new tests hit this.

File: lib/AST/TypeJoinMeet.cpp (hunk 1 above)
Comment (rudkx) [reply lib/AST/TypeJoinMeet.cpp]: As the comment for this member calls out, this makes it clear why the value is being returned.

**Comment by rudkx** [comment]
@swift-ci Please smoke test

**Comment by rudkx** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@rudkx I think this broke the source compatibility test suite:

```
Unimplemented type visitor!
UNREACHABLE executed at /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/swift/lib/AST/TypeJoinMeet.cpp:78!
0  swift                    0x000000010dd19788 llvm::sys::PrintStackTrace(llvm::raw_ostream&) + 40
1  swift                    0x000000010dd19e16 SignalHandler(int) + 422
2  libsystem_platform.dylib 0x00007fff5381ff5a _sigtramp + 26
3  libsystem_platform.dylib 0x00000001120722d4 _sigtramp + 3196396436
4  libsystem_c.dylib        0x00007fff5364a312 abort + 127
5  swift                    0x000000010dcafd20 LLVMInstallFatalErrorHandler + 0
6  swift                    0x000000010b5d649c (anonymous namespace)::TypeJoin::visitType(swift::CanType) + 28
7  swift                    0x000000010b5d6008 swift::CanType swift::CanTypeVisitor<(anonymous namespace)::TypeJoin, swift::CanType>::visit<>(swift::CanType) + 2264
8  swift                    0x000000010b5d559d (anonymous namespace)::TypeJoin::join(swift::CanType, swift::CanType) + 461
9  swift                    0x000000010b5d5388 swift::Type::join(swift::Type, swift::Type) + 56
10 swift                    0x000000010b0f4c36 swift::constraints::ConstraintSystem::PotentialBindings::addPotentialBinding(swift::constraints::ConstraintSystem::PotentialBinding, bool) + 614
11 swift                    0x000000010b0f3043 swift::constraints::ConstraintSystem::determineBestBindings() + 1827
12 swift                    0x000000010b156c61 swift::constraints::ConstraintSystem::solveSimplified(swift::constraints::Constraint*, llvm::SmallVectorImpl<swift::constraints::Solution>&, swift::FreeTypeVariableBinding) + 49
13 swift                    0x000000010b153d00 swift::constraints::ConstraintSystem::solveRec(llvm::SmallVectorImpl<swift::constraints::Solution>&, swift::FreeTypeVariableBinding) + 2320
14 swift                    0x000000010b15816d swift::constraints::DisjunctionChoice::solve(llvm::SmallVectorImpl<swift::constraints::Solution>&, swift::FreeTypeVariableBinding) + 61
15 swift                    0x000000010b15745e swift::constraints::ConstraintSystem::solveSimplified(swift::constraints::Constraint*, llvm::SmallVectorImpl<swift::constraints::Solution>&, swift::FreeTypeVariableBinding) + 2094
16 swift                    0x000000010b157f4b swift::constraints::Component::solve(swift::constraints::ConstraintSystem&, llvm::SmallVectorImpl<swift::constraints::Solution>&, swift::FreeTypeVariableBinding) + 667
17 swift                    0x000000010b15432d swift::constraints::ConstraintSystem::solveRec(llvm::SmallVectorImpl<swift::constraints::Solution>&, swift::FreeTypeVariableBinding) + 3901
18 swift                    0x000000010b15816d swift::constraints::DisjunctionChoice::solve(llvm::SmallVectorImpl<swift::constraints::Solution>&, swift::FreeTypeVariableBinding) + 61
19 swift                    0x000000010b15745e swift::constraints::ConstraintSystem::solveSimplified(swift::constraints::Constraint*, llvm::SmallVectorImpl<swift::constraints::Solution>&, swift::FreeTypeVariableBinding) + 2094
20 swift                    0x000000010b153d00 swift::constraints::ConstraintSystem::solveRec(llvm::SmallVectorImpl<swift::constraints::Solution>&, swift::FreeTypeVariableBinding) + 2320
21 swift                    0x000000010b15532d swift::constraints::ConstraintSystem::Candidate::solve(llvm::SmallDenseSet<swift::OverloadSetRefExpr*, 4u, llvm::DenseMapInfo<swift::OverloadSetRefExpr*> >&) + 1245
22 swift                    0x000000010b1560fb swift::constraints::ConstraintSystem::shrink(swift::Expr*) + 331
23 swift                    0x000000010b15680e swift::constraints::ConstraintSystem::solve(swift::Expr*&, swift::Type, swift::ExprTypeCheckListener*, llvm::SmallVectorImpl<swift::constraints::Solution>&, swift::FreeTypeVariableBinding) + 462
24 swift                    0x000000010b1e37e4 swift::TypeChecker::solveForExpression(swift::Expr*&, swift::DeclContext*, swift::Type, swift::FreeTypeVariableBinding, swift::ExprTypeCheckListener*, swift::constraints::ConstraintSystem&, llvm::SmallVectorImpl<swift::constraints::Solution>&, swift::OptionSet<swift::TypeCheckExprFlags, unsigned int>) + 52
25 swift                    0x000000010b1e6f19 swift::TypeChecker::typeCheckExpression(swift::Expr*&, swift::DeclContext*, swift::TypeLoc, swift::ContextualTypePurpose, swift::OptionSet<swift::TypeCheckExprFlags, unsigned int>, swift::ExprTypeCheckListener*, swift::constraints::ConstraintSystem*) + 969
26 swift                    0x000000010b278285 swift::ASTVisitor<(anonymous namespace)::StmtChecker, void, swift::Stmt*, void, void, void, void>::visit(swift::Stmt*) + 6597
27 swift                    0x000000010b27698d swift::ASTVisitor<(anonymous namespace)::StmtChecker, void, swift::Stmt*, void, void, void, void>::visit(swift::Stmt*) + 205
28 swift                    0x000000010b274c4d swift::TypeChecker::typeCheckFunctionBodyUntil(swift::FuncDecl*, swift::SourceLoc) + 269
29 swift                    0x000000010b275851 swift::TypeChecker::typeCheckAbstractFunctionBody(swift::AbstractFunctionDecl*) + 353
30 swift                    0x000000010b2968fa typeCheckFunctionsAndExternalDecls(swift::TypeChecker&) + 282
31 swift                    0x000000010b297886 swift::performTypeChecking(swift::SourceFile&, swift::TopLevelContext&, swift::OptionSet<swift::TypeCheckingFlags, unsigned int>, unsigned int, unsigned int, unsigned int, unsigned int) + 1862
32 swift                    0x000000010aed7ec7 swift::CompilerInstance::parseAndCheckTypes(swift::CompilerInstance::ImplicitImports const&) + 839
33 swift                    0x000000010aed7752 swift::CompilerInstance::performSema() + 466
34 swift                    0x000000010a31734f performCompile(swift::CompilerInstance&, swift::CompilerInvocation&, llvm::ArrayRef<char const*>, int&, swift::FrontendObserver*, swift::UnifiedStatsReporter*) + 1887
35 swift                    0x000000010a315bde swift::performFrontend(llvm::ArrayRef<char const*>, char const*, void*, swift::FrontendObserver*) + 3310
36 swift                    0x000000010a2d3ed3 main + 2051
37 libdyld.dylib            0x00007fff5359e115 start + 1
Stack dump:
0.	Program arguments: /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/build/compat_macos/install/toolchain/usr/bin/swift -frontend -c /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/UIKit/UITableViewExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/ArrayExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/DataExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/FloatExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/UIKit/UISwitchExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/UIKit/UITextViewExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/UIKit/UINavigationBarExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/UIKit/UISliderExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/UIKit/UIImageExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/DictionaryExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/UIKit/UIBarButtonItemExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/UIKit/UINavigationControllerExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/SwifterSwift.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/UIKit/UIButtonExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/Cocoa/NSAttributedStringExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/LocaleExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/DoubleExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/UIKit/UIViewExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/URLExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/BoolExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/CharacterExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/DateExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/UIKit/UICollectionViewExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/IntExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/Cocoa/CGColorExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/UIKit/UITextFieldExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/UIKit/UILabelExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/UIKit/UISearchBarExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/UIKit/UIColorExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/UIKit/UIViewControllerExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/CollectionExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/UIKit/UIImageViewExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/OptionalExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/Cocoa/CGPointExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/Cocoa/CGFloatExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/UIKit/UITabBarExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/UIKit/UINavigationItemExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/UIKit/UISegmentedControlExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/Cocoa/CGSizeExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/UIKit/UIAlertControllerExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/UIKit/UIStoryboardExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/StringExtensions.swift /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/Cocoa/CLLocationExtensions.swift -supplementary-output-file-map /var/folders/_8/79jmzf2142z2xydc_01btlx00000gn/T/supplementaryOutputs-88867a -target armv7-apple-ios8.0 -enable-objc-interop -sdk /Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.3.sdk -I /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Products/Release-iphoneos -F /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Products/Release-iphoneos -g -import-underlying-module -module-cache-path /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/ModuleCache.noindex -profile-generate -profile-coverage-mapping -swift-version 3 -O -serialize-debugging-options -Xcc -I/Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/swift-overrides.hmap -Xcc -iquote -Xcc /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/SwifterSwift-generated-files.hmap -Xcc -I/Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/SwifterSwift-own-target-headers.hmap -Xcc -I/Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/SwifterSwift-all-non-framework-target-headers.hmap -Xcc -ivfsoverlay -Xcc /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/all-product-headers.yaml -Xcc -iquote -Xcc /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/SwifterSwift-project-headers.hmap -Xcc -I/Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Products/Release-iphoneos/include -Xcc -I/Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/DerivedSources/armv7 -Xcc -I/Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/DerivedSources -Xcc -ivfsoverlay -Xcc /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/unextended-module-overlay.yaml -Xcc -working-directory/Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift -module-name SwifterSwift -num-threads 16 -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/UITableViewExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/ArrayExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/DataExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/FloatExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/UISwitchExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/UITextViewExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/UINavigationBarExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/UISliderExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/UIImageExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/DictionaryExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/UIBarButtonItemExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/UINavigationControllerExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/SwifterSwift.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/UIButtonExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/NSAttributedStringExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/LocaleExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/DoubleExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/UIViewExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/URLExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/BoolExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/CharacterExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/DateExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/UICollectionViewExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/IntExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/CGColorExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/UITextFieldExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/UILabelExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/UISearchBarExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/UIColorExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/UIViewControllerExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/CollectionExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/UIImageViewExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/OptionalExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/CGPointExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/CGFloatExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/UITabBarExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/UINavigationItemExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/UISegmentedControlExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/CGSizeExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/UIAlertControllerExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/UIStoryboardExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/StringExtensions.o -o /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/build/Build/Intermediates.noindex/SwifterSwift.build/Release-iphoneos/SwifterSwift iOS.build/Objects-normal/armv7/CLLocationExtensions.o 
1.	While type-checking 'average()' at /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/CollectionExtensions.swift:66:9
2.	While type-checking expression at [/Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite/project_cache/SwifterSwift/Source/Extensions/CollectionExtensions.swift:68:10 - line:68:73] RangeText="isEmpty ? 0 : Double(reduce(0, +)) / Double(endIndex-startIndex)"

```

**Comment by rudkx** [comment]
As you mentioned elsewhere this didn't cause the break (we already had missing cases).

I addressed the issue here: https://github.com/apple/swift/pull/15448

**Pull request merged**